### ISSUE TITLE
New Tag for 03Feb2017 samples

### ISSUE THREAD

**Issue by alefisico** [cmsb2g/B2GAnaFW#71]
Hi all
not only because of this: https://hypernews.cern.ch/HyperNews/CMS/get/b2g-selections/332.html
but I think is time for the next tag. Is there something extra that we want to include in this tag? 
If I remember correctly, we didn't launch a big campaign for the last tag v8.0.x_v2.5.

**Comment by jkarancs** [comment]
Dear Alejandro,
Can we add my PR (#70)?
https://github.com/cmsb2g/B2GAnaFW/pull/70

This fixes a problem with the stochastic part of jet smearing.

**Comment by skyriacoCMS** [comment]
Hello all,

So, I have made a new branch named CMSSW_8_0_X_V3 at the forked repository
I have of the B2GAnaFW.

https://github.com/skyriacoCMS/B2GAnaFW/tree/CMSSW_8_0_X_V3/src

0 )This new branch includes all the latest changes made to the tag
/v8.0.x_v2.5  and starts with that as a base.

1 )The long awaited Electron MVA value is now stored in the electron
collection.

2) Photon IDs updated to the correct ones

3) PhotonJets Collection is now part of the AK8CHS collection. This
collection from the beggining was some extra variables for the jet
collections and I was storing a reference to the jet that was associated
with the valeus stored with an index. This was problematic. So now I made
it part of the standard AK8CHS collection.  The size of it does not
increases actually one variable gets eliminated - the index.

4) For the issue Jannos Reports, I tried to incorporate it to my branch but
I get errors when I try to compile. - It is a single line  and this is very
strange.


How should I proceed? I can also make a pull request for the new branch I
made and also add Jannos change once we figure out why I get compilation
error out of it.

Do you need to make a branch first at the cmsb2g repo with the same name
and my pull request targets that branch?

Cheers,
Savvas Kyriacou


*Savvas Kyriacou*
*Department of Physics and Astronomy         *
*Rutgers, The State University of New Jersey *




*CERN B8 R-020       ,__,           (o.o)           /),,,,)            "  "
      *

**Comment by alefisico** [comment]
Hi all
@jkarancs is that the only fix that you want to include? Unfortunately I dont have writing permissions, but maybe @dmajumder can help with that? (Actually maybe it will be useful for someone else to have the permissions? in case Devdatta is busy)
@skyriacoCMS and @jkarancs, let me check if I can solve this compilation problem incorporating both changes, and I'll go back to you.
Also, is there a strict timeline for this new tag? i.e., is it someone that needs this right now? (just to know)

**Comment by jkarancs** [comment]
@alefisico 
Yes, that's the only change. It's just one line, so someone with write permission should do it. I made a PR because I don't have such permission, but it seems noone accepted it yet.

**Comment by dmajumder** [comment]
I just commented on the PR, sorry for some reason I missed this notification in my mailbox.

As for the tag for 03Feb, I am not sure if it is urgent. Do one of you want to use it for your analyses right away? I am slightly concerned about the space, we may need to move things around or delete old stuff.

**Comment by alefisico** [comment]
I think not from our side, maybe we can ask in HN?

**Comment by dmajumder** [comment]
I am waiting for some clarification from Janos before merging #70 
I think this PR can go into CMSSW_8_0_X_V3 along with Savvas's new additions.

Another thing
@alefisico For AK8CHS jets, I think the matched Puppi subjet indexes are not being stored properly, r being stored at all. Can you check this part?

**Comment by jkarancs** [comment]
@dmajumder 
I don't see your comment, but let me clarify the change in PR #70.
The JER twiki has a recipe for stochastic smearing, which is the 3rd formula here:
https://twiki.cern.ch/twiki/bin/viewauth/CMS/JetResolution#Smearing_procedures

sigma_JER is written as the "relative pt resolution", so the ratio of the absolute jet resolution (which is what we have currently) divided by the pt of the jet. I made this correction in the PR, it's just one line:
https://github.com/cmsb2g/B2GAnaFW/pull/70/commits/2258c2da84265056613ee755d4cc8cba93799f66#diff-f49af754dc51e5242da678a17c234588L253

So to summarize, this third formula is going to be implemented like this after the fix:
https://github.com/jkarancs/B2GAnaFW/blob/2258c2da84265056613ee755d4cc8cba93799f66/src/JetUserData.cc#L253-L254

**Comment by dmajumder** [comment]
Summary: #70 is closed and we do not need the change.
Once I hear from @alefisico about the storage of the AK8PUPPI subjet indices matched to the AK8CHS jets we can move forward

**Comment by skyriacoCMS** [comment]
@dmajumder Hello,  If you create a new branch on the official framework I can make a pull request with my changes - which are a lot - and then on top of that add the indices issue.

**Comment by dmajumder** [comment]
@skyriacoCMS I have created a new branch https://github.com/cmsb2g/B2GAnaFW/tree/CMSSW_8_0_X_V3
You can make the PR there

**Comment by alefisico** [comment]
btw, I will check the subjet indeces once #72 is merged.

**Comment by dmajumder** [comment]
#72 Merged. With these developments, any production needs to re-produce both the data and the MC. Also, did anybody change the GT for 03Feb re-miniaod?

**Comment by alefisico** [comment]
> Also, did anybody change the GT for 03Feb re-miniaod?

I dont think so, but I can do it once I commit the changes in the indexes.

**Comment by dmajumder** [comment]
let's have a look once we have the subjet indices checked.

**Comment by alefisico** [comment]
@dmajumder the subjet indices are not stored because the subjets are not stored. Remember that we are talking about the Puppi jets link to the CHS jets directly in miniAOD. But do we want those subjets? I think we have an entire collection of Puppi jets with subjets and all, so maybe it is not needed. 
On the other side I updte the JECs and all the commands needed. I can push my branch with those changes if you want.

**Comment by dmajumder** [comment]
@alefisico You're right. However, one reason is to apply the PUPPI SD mass correction for W/Z tagging. This requires the uncorrected PUPPI SD mass. So I'd suggest storing this as a userFloat. 

And yes, please push when you're done.

**Comment by alefisico** [comment]
@dmajumder to store the four momentum of the subjets as userFloat? (just to be sure)

**Comment by dmajumder** [comment]
@alefisico Only the PUPPI SD uncorrected mass should suffice. You usually calculate it using the sum of the subjet uncorrected four-momenta.

Unfortunately we we have no way of knowing how these recipes will evolve in the future, so this is just an interim solution.

**Comment by alefisico** [comment]
sorry for the delay but I had some stupid merging conflicts that I had to solve. 
I just create #73, after that I think we are ready.

**Comment by dmajumder** [comment]
PR#73 merged. Do some tests before a new tag?

**Comment by skyriacoCMS** [comment]
Hello All,

I did a new pull request with minimal changes to the PhotonJets collection.
I think this should be included in the new tag. I could do the tests that
we need to do for the new tag.

Cheers,


*Savvas Kyriacou*
*Department of Physics and Astronomy         *
*Rutgers, The State University of New Jersey *




*CERN B8 R-020       ,__,           (o.o)           /),,,,)            "  "
      *

**Comment by dmajumder** [comment]
I have merged #74 
time to do tests. BTW, given that we are adding several novel stuff, I think we need a light validation framework to test some variables before making production tags. Any ideas?

**Comment by jkarancs** [comment]
Dear Devdatta,

Please hold on for a little bit with the new tag, I'm testing the
uncorrected SoftDrop mass, it is different to what I get in my calculation.
I suspect Ale did not exactly match the CHS jet to the Puppi one finding
the closest distance, just accepted the first which was within the cone. I
try to check if this is true and report back in a few minutes.

Bests,
Janos

**Comment by jkarancs** [comment]
Dear all,

In the mean time let me elaborate.
I already privately calculated the uncorrected puppi softdrop mass (needed for W tagging) the same way as it is implemented currently, with some exceptions:
- I was minimizing the matching distance and only accept the closest jet
- I was using "slimmedJetsAK8" collection (which is available in MINIAOD) and their "SoftDropPuppi" subjets, exactly like in the recipe [1]. 
We currently use: "packedPatJetsAK8PFPuppiSoftDrop" which is not in MINIAOD and its subjet collection: "SoftDrop".

What do you think? this is the relevant code.

[1] - https://github.com/cms-jet/PuppiSoftdropMassCorr/#get-uncorrected-puppi-soft-drop-mass-from-miniaod
[2] - https://github.com/cmsb2g/B2GAnaFW/blob/CMSSW_8_0_X_V3/src/BoostedJetToolboxUserData.cc#L86-L99

**Comment by jkarancs** [comment]
Dear all,

Additional to the things I mentioned previously, the subjet 4 momentum also seemed wrong. We need to use the correctedP4(0) according to the POG recipe.
I decided to make a PR (#76) and also add the MVA electron values/categories for general purpose and HZZ IDs, I think others might find them useful.

**Comment by jkarancs** [comment]
Using the L1L2L3 corrected subjets when computing softdrop mass is also useful (for top tagging) so I added back that variable too. Now the two new variables will be suitable for both the latest recommendations for W and top tagging. For the former one also need to apply the GEN/RECO corrections, but this can change so it's better to be applied on-the-fly in the analyses.

**Comment by dmajumder** [comment]
@jkarancs we already store the SD mass so what extra variables are needed?

We are not storing any corrections since as you said, they change. We just want to store the minimal set of variables needed to compute these corrections

**Comment by alefisico** [comment]
Hi all (sorry for the late reply, I was very sick the last week)
@jkarancs about 
> I was using "slimmedJetsAK8" collection (which is available in MINIAOD) and their "SoftDropPuppi" subjets, exactly like in the recipe [1].
> We currently use: "packedPatJetsAK8PFPuppiSoftDrop" which is not in MINIAOD and its subjet collection: "SoftDrop".

The reason of using the packedPatJetsAK8PFPuppiSoftDrop instead of the slimmedJetsAK8 is because we are NOT using the slimmedJetsAK8 jets at all. We are still reclustering everything througth the jetToolbox for AK8 jets and mimic the puppi jets in the CHS collection as in miniAOD. Maybe that is the reason your calculations are different, although I will keep your way to make sure the AK8 jet is the closest to the jet with subjets.

**Comment by jkarancs** [comment]
@dmajumder
Indeed we have the Puppi softdrop mass already, but we don't exactly use the same corrections as the POG. They checked different corrections, and they currently recommend on their twiki [1] to use "groomed top jets (SoftDrop) and AK4 L1L2L3 JEC corrections applied to the subjets", the same as on second bullet on slide 4 here [2]. Our Puppi SoftDrop mass is somewhat different, so to get the exact same results I suggest to use this new corrSDMassPuppi variable.

[1] - https://twiki.cern.ch/twiki/bin/view/CMS/JetTopTagging#13_TeV_working_points_CMSSW_8_0
[2] - https://indico.cern.ch/event/616638/contributions/2491478/attachments/1420668/2176938/TopTaggingSFMeasurement.pdf

**Comment by jkarancs** [comment]
@alefisico 
I wish you to get well soon.

This new calculation of "uncorrSDMassPuppi" using the MINIAOD collection is only there in order to exactly get the same uncorrected mass that the POG uses for W tagging, otherwise we introduce some kind of difference which is in most cases small, but in some others it is not negligible. Of course we are free to define and use our own reclustered collections, but in my opinion, for mass tagging (both W and top), the matching should be done using the MINIAOD collection in order to be able to reproduce the POG results and use their corrections/scale factors.

**Comment by skyriacoCMS** [comment]
@jkarancs 

Hello All, 

So regarding the GP MVA IDs ok, fair enough, the idea was to just have a single variable - the score of the mva  and then the user decides what to cut on. The WP are on the EG twikis but mostly we have seen (IN THE b2g group -Dylan Rankin) that this new training WPs was not doing as good as the old one and one had to shift the WP to get similar or better s/b and stability in higher pt electrons. So I thought to just add the mva value and each one decides what suites them. But what you have is ok in anycase. 

 Good that you added the HZZ MVA. I did not know that people wanted this so I did not add it. 

For the PhotonJets - Sorry for this. Its a typo - single line fix. I will fix it and make a PR once you guys settle on the rest of the issues. 

Cheers

**Comment by jkarancs** [comment]
@skyriacoCMS
The HZZ MVA ID is indeed useful for people who want to a) have a very high efficiency selection, or b) veto electrons or c) want to extend below 10GeV pt, so I think this is good ID to keep also.

**Comment by dmajumder** [comment]
@alefisico @skyriacoCMS @jkarancs are we ready to create a new tag? 
I've got queries from some people, so this is now a bit high priority.

**Comment by alefisico** [comment]
I dont have any other thing to include from my side.

**Comment by dmajumder** [comment]
@alefisico thanks. once the other two confirm i think we're good to go.

**Comment by skyriacoCMS** [comment]
@dmajumder @jkarancs @alefisico 
Hello guys, 

From my side I have nothing more to add either

**Comment by jkarancs** [comment]
@dmajumder 
Also ready from my side. Time to make new ntuples!

**Comment by dmajumder** [comment]
I've created a new release, please check the notes and the changelog w.r.t. the previous production release:
https://github.com/cmsb2g/B2GAnaFW/releases/tag/v8.0.x_v3.1

I'll announce it tomorrow

**Comment by dmajumder** [comment]
@decosa @oiorio 

Sorry but could you summarize the email thread here? It's easier to keep track.

I am still not sure if MET recipe needs to be updated. It is mentioned here
https://twiki.cern.ch/twiki/bin/view/CMS/MissingETRecipes
but not here apparently.
https://twiki.cern.ch/twiki/bin/view/CMSPublic/ReMiniAOD03Feb2017Notes#MET_Recipes

BTW, I am not sure if we all need the muon and EG cleaned MET. We should still keep the Type 1 corrected MET since that should satisfy most needs. 

Please comment and we'll follow up with a new tag.

**Comment by cmsb2g** [comment]
Hello,

The summary is the following:

1) Right now only muon corrected MET is stored in the ntuples
2) Only one of the two merge from MET recipe has been added to the README, should we include both of them?




   git cms-merge-topic cms-met:METRecipe_8020 -u


   git cms-merge-topic cms-met:METRecipe_80X_part2 -u

However, the second one is needed only to recompute the gain switch e/gamma propagation (I asked Zeynep for confirmation).

So, my suggestion is to go ahead with the production, given that we are already really late, and ask if anyone need muon+egamma corrected MET.
In case, we can go ahead with another tag.

Cheers,

Annapaola




On 11 May 2017, at 01:30, dmajumder <notifications@github.com<mailto:notifications@github.com>> wrote:


@decosa<https://github.com/decosa> @oiorio<https://github.com/oiorio>

Sorry but could you summarize the email thread here? It's easier to keep track.

I am still not sure if MET recipe needs to be updated. It is mentioned here
https://twiki.cern.ch/twiki/bin/view/CMS/MissingETRecipes
but not here apparently.
https://twiki.cern.ch/twiki/bin/view/CMSPublic/ReMiniAOD03Feb2017Notes#MET_Recipes

BTW, I am not sure if we all need the muon and EG cleaned MET. We should still keep the Type 1 corrected MET since that should satisfy most needs.

Please comment and we'll follow up with a new tag.

**Comment by dmajumder** [comment]
ok, I will publicise the tag then, and in the next round we'll see if other METs are needed.
We really have to get the object contacts to sign off from now on.

**Comment by dmajumder** [comment]
[v8.0.x_v3.2](https://github.com/cmsb2g/B2GAnaFW/releases/tag/v8.0.x_v3.2) made. I am going to announce it

**Comment by dittmer** [comment]
Sorry to add this as a late comment (and coming from outside the group), but does the latest tag include the [EGM regressions](https://twiki.cern.ch/twiki/bin/viewauth/CMS/EGMRegression)?

**Comment by dmajumder** [comment]
replied here https://hypernews.cern.ch/HyperNews/CMS/get/b2g-selections/337.html
Closing this issue.
